things/lift/modbus/js/main.js

```diff
@@ -0,0 +1,179 @@
+        "readproperty"
+    ],
+    "modbus:entity": "Coil",
+    "modbus:function": "readCoil",
+    "contentType": "application/octet-stream"
+}, {
+    "href": `${thingUnitID}?address=1&quantity=1`,
+    "op": [
+        "writeproperty"
+    ],
+    "modbus:entity": "Coil",
+    "modbus:function": "writeSingleCoil",
+    "contentType": "application/octet-stream"
+}]
+
+
+thingDescription['properties']['lightSwitch']['forms'] = lightSwitchForms
+
+const onTheMoveForms = [{
+    "href": `${thingUnitID}?address=1&quantity=1`,
+    "op": [
+        "readproperty"
+    ],
+    "modbus:entity": "DiscreteInput",
+    "modbus:function": "readDiscreteInput",
+    "contentType": "application/octet-stream"
+}]
+
+discreteInputs[0] = 0
+thingDescription['properties']['onTheMove']['forms'] = onTheMoveForms
+
+const floorNumberForms = [{
+    "href": `${thingUnitID}?address=1&quantity=1`,
+    "op": [
+        "readproperty"
+    ],
+    "modbus:entity": "HoldingRegister",
+    "modbus:function": "readHoldingRegisters",
+    "contentType": "application/octet-stream"
+}, {
+    "href": `${thingUnitID}?address=1&quantity=1`,
+    "op": [
+        "writeproperty"
+    ],
+    "modbus:entity": "HoldingRegister",
+    "modbus:function": "writeSingleHoldingRegister",
+    "contentType": "application/octet-stream"
+}]
+
+thingDescription['properties']['floorNumber']['forms'] = floorNumberForms
+
+const vector = {
+    getDiscreteInput: function(addr, unitID) {
+        if (thingUnitID === unitID) {
+            console.log(`Reading discrete input @${addr}`)
+            return discreteInputs[addr - 1];
+        }
+    },
+    getInputRegister: function(addr, unitID) {
+        if (thingUnitID === unitID) {
+            return inputRegisters[addr - 1];
+        }
+    },
+    getHoldingRegister: function(addr, unitID, callback) {
+        if (thingUnitID === unitID) {
+            setTimeout(function() {
+                callback(null, holdingRegisters[addr - 1])
+            }, 10)
+        }
+    },
+    getCoil: function(addr, unitID) {
+        return new Promise(function(resolve) {
+            setTimeout(function() {
+                resolve(coils[addr - 1])
+            }, 10)
+        })
+    },
+    setRegister: function(addr, value, unitID) {
+        if (thingUnitID === unitID) {
+            // trying to change floor number
+            if (addr === 1) {
+                // lift is on the move
+                if (discreteInputs[0]) {
+                    console.log("Lift is on the move, cannot change the floor number")
+                } else {
+                    console.log(`Changing the floor number to ${value}`)
+                    holdingRegisters[addr - 1] = value
+                    discreteInputs[0] = 1
+                    setTimeout(() => {
+                        discreteInputs[0] = 0
+                    }, 5000)
+                }
+            }
+        }
+
+        return
+    },
+    setCoil: function(addr, value, unitID) {
```

Comment:
a log of the change would be nice

Reply:
Fixed at [1e7bc99](https://github.com/eclipse-thingweb/test-things/pull/9/commits/1e7bc99a0e815868f9254edfff03ef9562635cd2)